### ISSUE TITLE
Investigate loading of SmartDaqApplication implementations

### ISSUE THREAD

**Issue by eflumerf** [DUNE-DAQ/appmodel#95]
We use `cetlib` to load plugins with well-defined APIs elsewhere in dune-daq, so it should be possible to have generate_modules implementations be loaded at runtime via this plugin mechanism. This would allow SmartDaqApplication implementations to live in subsystem repositories (e.g. `timinglibs`) as well as subsystem-specific schemas.

**Comment by eflumerf** [comment]
Possibly resolved by #197?

**Comment by gcrone** [comment]
> Possibly resolved by [#197](https://github.com/DUNE-DAQ/appmodel/pull/197)?

Yes, I can confirm that the `generate_modules` method from a dal library in an independent package does get called from the  `generate_modules_test` and `daq_application` applications.
